webui/src/components/SearchForm.vue

```diff
@@ -0,0 +1,100 @@
+<template>
+    <div class="input-form input-search">
+        <input type="text"
+               v-model="text"
+               placeholder="Find people"
+               @input="searchSubjects(text)"
+               @focus="showSuggestedSubject"
+               @blur="hideSuggestedSubject"/>
+        <div class="action-button" @click="searchSubjects(text)">
+            <SearchIcon/>
+        </div>
+
+        <transition name="fade">
+            <div class="suggested-subjects-list" v-show="suggestedSubjectShoved">
+                <div v-for="item in suggestedSubjects"
+                     class="suggested-subjects-item"
+                     @click="selectInterestingSubject(item)">{{item.displayName}}</div>
+            </div>
+        </transition>
+    </div>
+</template>
+
+<script>
+    import SearchIcon from './../components/Icons/SearchIcon'
+
+    export default {
+        name: "SearchForm",
+
+        components: {SearchIcon},
+
+        data() {
+            return {
+                suggestedSubjectShoved: false,
+            }
+        },
+
+        computed: {
+            text: {
+                get() {
+                    return this.$store.state.user.searchValue;
+                },
+
+                set(changed) {
+                    this.$store.commit('CHANGE_SEARCH_VALUE', changed);
+                }
+            },
+
+            suggestedSubjects: {
+                get() {
+                    return this.$store.state.user.suggestedSubjectsList;
```

Comment:
This is kind of subjective, but I would prefer to use getters whenever possible. For example some thoughts are here https://stackoverflow.com/questions/47191231/why-is-state-visible-to-components